### PR TITLE
Added type definitions for elastic-apm-node

### PR DESCRIPTION
**Pull request by Shahaed**
Adding type definitions for this package: https://github.com/elastic/apm-agent-nodejs.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/elastic-apm-node/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "adjacent-overload-signatures": false,
```
[the remark this answers is not in the document]
Comment (Shahaed) [reply types/elastic-apm-node/tslint.json]: I copied and pasted one from the repo. Is there a standard linter?

[the remark this answers is not in the document]
Comment (amcasey) [reply types/elastic-apm-node/tslint.json]: I believe if you just leave out the rules section, it will run all the default rules (based on the extends).

**Comment by Shahaed** [comment]
@amacleay Updated the linter rules and simplified `index.d.ts`. I can squash the commits after you review

**Comment by amacleay** [comment]
I presume you meant @amcasey,  @Shahaed ?

**Review by amcasey — APPROVED** [review]

**Comment by amcasey** [comment]
Thanks, @amacleay!  Sorry for the confusion.

File: types/elastic-apm-node/tslint.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "extends": "dtslint/dt.json",
```
Comment (amcasey) [inline types/elastic-apm-node/tslint.json]: I'm guessing the build is unhappy about this trailing comma.

Comment (Shahaed) [reply types/elastic-apm-node/tslint.json]: Got rid of it and squashed! Thanks for the approval.

**Comment by Shahaed** [comment]
Made requested changes! Should be ready to merge @amcasey

**Pull request merged**